### ISSUE TITLE
Backstopjs hang at the end of a run

### ISSUE THREAD

**Issue by yuriydalekyy** [garris/BackstopJS#736]
Hi.
Sometimes, backstop can't finish work and i can't have report.

report | Test completed...
       report | 89 Passed
       report | 5 Failed
       report | Writing jUnit Report
       report | Writing browser report
       report | jUnit report written to: backstop_data\ci_report\xunit.xml
       report | Resources copied
       report | Copied configuration to: C:\For Run Jenkins\workspace\backstopJS\backstop_data\html_report\config.js
      COMMAND | Executing core for `openReport`
   openReport | Opening report.
       report | *** Mismatch errors found ***
      COMMAND | Command `report` ended with an error after [41.072s]
      COMMAND | Error: Mismatch errors found.
                    at C:\Users\Andrey\AppData\Roaming\npm\node_modules\backstopjs\core\command\report.js:113:17
                    at <anonymous>
                    at process._tickCallback (internal/process/next_tick.js:160:7)
      COMMAND | Command `test` ended with an error after [623.739s]
      COMMAND | Error: Mismatch errors found.
                    at C:\Users\Andrey\AppData\Roaming\npm\node_modules\backstopjs\core\command\report.js:113:17
                    at <anonymous>
                    at process._tickCallback (internal/process/next_tick.js:160:7)
............

It is all..
Can you help me?

**Comment by pbb72** [comment]
I have similar problems. I do get a report, but the script hangs at the end. One thing I noticed was the messages _before_ the "Test completed" message:

```
      compare | Chromy error: Error. See scenario – xxx (mobile)
      compare | Chromy error: Error. See scenario – xxx (desktop)
      compare | Chromy error: Error. See scenario – yyy (mobile)
      compare | Chromy error: Error. See scenario – yyy (desktop)
```

For the scenarios that errored, no screenshots are generated.

**Comment by yuriydalekyy** [comment]
I haven't such problem in my logs @pbb72

**Comment by pbb72** [comment]
This is my problem with hanging: #737

**Comment by garris** [comment]
Hi all, we are seeing some issues with some chromy instances failing (often when a selector is missing or there is some other unexpected condition) — I would recommend ensuring you have the latest version and use the `engine: puppeteer ` option.  See the docs for more info — this version is much more stable.

**Comment by garris** [comment]
Also see here — https://github.com/garris/BackstopJS/blob/master/README.md#chrome-zombies

**Comment by garris** [comment]
This problem should be completely resolved using puppeteer engine with backstopjs@3.2.15.   Please reopen if there are still issues.
